src/routes/route.py

```diff
@@ -1,67 +1,95 @@
 
 from data_loader.population.utils import NODE_ID, POPULATION
 from routes.route_utils import path
 
 
 class Route:
-    def __init__(self, route_path: path, num_people_on_route: int) -> None:
+    def __init__(
+        self, route_path: path, num_people_on_route: int, departure_times: list[int]
+    ) -> None:
+        if len(departure_times) != num_people_on_route:
+            logging.fatal(
+                "Number of departure times must equal the number of people on route."
+            )
         self.path = route_path
         self.num_people_on_route = num_people_on_route
-        self.departure_times: dict[int, int] = {0: num_people_on_route}
+        self.departure_times = departure_times
 
-    def introduce_departure_times(self, chunks: int, interval: int) -> None:
-        """
-        Introduces different departure times for the persons taking the given route.
 
-        :param chunks: In how many different chunks should the persons on the route depart.
-        :param interval: How long should the interval between each departure be. Given in seconds.
-        :return: A dictionary of departure times where keys are seconds after midnight and values are the number of people departing on that time.
-        """
-        departure_times: dict[int, int] = {}
-        people_per_interval = int(self.num_people_on_route / chunks)
-        remaining_people = self.num_people_on_route % chunks
+def _departure_times(total_population: int, start: int, end: int) -> NDArray[np.int_]:
+    """
+    Introduces different departure times for the persons taking the given route, spread on a normal distribution
+    between the given start and end times.
+    :param total_population: The total population.
+    :param start: start of normal distribution. Given in seconds.
+    :param end: end of normal distribution. Given in seconds.
+    """
+    mean = (start + end) / 2  # mean
+    std_dev = (end - start) / 6  # Approx. 99.7% of values within range
+    rng = np.random.default_rng()
+    departures = rng.normal(loc=mean, scale=std_dev, size=total_population).astype(
+        np.int_
+    )
+    return departures
 
-        seconds_from_midnight = 0
-        if chunks < self.num_people_on_route:
-            for i in range(chunks):
-                count = people_per_interval + (
-                    1 if i < remaining_people else 0
-                )  # Distribute remainder evenly
-                departure_times[seconds_from_midnight] = count
-                seconds_from_midnight += interval
-        else:  # There are more chunks than people, so there will be 1 person in the first chunk(s) and then 0 in the remaining. The remaining will not even be present in the departure_times dict.
-            for i in range(self.num_people_on_route):
-                departure_times[seconds_from_midnight] = 1
-                seconds_from_midnight += interval
 
-        self.departure_times = departure_times
+def _get_num_people_on_route(
```

Comment:
Is this function placed correctly? It is used to define the num people on route which is a field on the class. Maybe im thinking too object oriented here though. Just find it a bit weird that its placed in this file when it needs to be called before a Route object can be made

Reply:
Nevermind, just saw the create method below. It's fine! Thought it was called from outside this file